File: v1_12/src/main/java/me/fullidle/ficore/ficore/v1_12/MoveWrapper.java
package me.fullidle.ficore.ficore.v1_12;

import com.pixelmonmod.pixelmon.battles.attacks.AttackBase;
import me.fullidle.ficore.ficore.common.api.pokemon.wrapper.IMoveWrapper;

public class MoveWrapper extends IMoveWrapper<AttackBase> {
    public MoveWrapper(AttackBase original) {
        super(original);
    }

    @Override
    public String getName() {
        return getOriginal().getAttackName();
    }

    @Override
    public Class<AttackBase> getType() {
        return AttackBase.class;
    }
}

File: v1_21/Cob/src/main/kotlin/me/fullidle/ficore/ficore/vCob_21/MoveWrapper.kt
package me.fullidle.ficore.ficore.vCob_21

import com.cobblemon.mod.common.api.moves.Move
import me.fullidle.ficore.ficore.common.api.pokemon.wrapper.IMoveWrapper

class MoveWrapper(original: Move) : IMoveWrapper<Move>(original) {
    override fun getName(): String {
        return original.name
    }

    override fun getType(): Class<Move> {
        return Move::class.java
    }
}
